sw/vendor/riscv_compliance/riscv-target/opentitan/device/rv32imc/wrap.c

```diff
@@ -0,0 +1,34 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <string.h>
+
+#include "common.h"
+#include "uart.h"
+
+#define SIM_TERM_ADDR 0x10008000
+
+void dump_signature(void) {
+  extern uint32_t begin_signature[];
+  extern uint32_t end_signature[];
+
+  uint32_t size = end_signature - begin_signature;
+  uart_init(UART_BAUD_RATE);
+  // uart_send_str("PASS!\r\n");
+  for (uint32_t i = 0; i < size; i++) {
+    uart_send_str("SIG: ");
+    uart_send_uint(REG32(begin_signature + i), 32);
+    uart_send_str("\r\n");
+  }
+
+  uart_send_str("PASS!\r\n");
```

Comment:
Is there a condition that causes FAIL! to be output instead? What is the success condition of this test?

Reply:
This test just outputs the compliance signature and terminates.  Something else checks the signature for success or failure.  Or the test going off into the weeds would be failure too.

Reply:
yeah if the test goes into the weeds and never returns anything, pytest will timeout after 60 seconds and register a failure. Of course in this case the log will be wrong, so the reference compare fails too.